src/main/scala/cromwell/engine/backend/jes/JesAttributes.scala

```diff
@@ -3,51 +3,73 @@ package cromwell.engine.backend.jes
 import java.net.URL
 
 import com.typesafe.config.ConfigFactory
+import cromwell.authentication.oauth2.{ClientSecrets, SimpleClientSecrets}
+import cromwell.authentication.{DockerConfiguration, DockerCredentials}
+import cromwell.engine.backend.jes.authentication.{JesDockerCredentials, JesAuthInformation, JesAuthMode}
 import cromwell.util.ConfigUtil._
 
 import scala.language.postfixOps
 import scalaz.Scalaz._
+import scalaz.Validation.FlatMap._
 import scalaz._
 
 case class JesAttributes(applicationName: String,
                          project: String,
                          executionBucket: String,
                          endpointUrl: URL,
-                         authMode: GcsAuthMode,
-                         dockerCredentials: Option[DockerCredentials])
+                         authMode: JesAuthMode,
+                         dockerCredentials: Option[JesDockerCredentials],
+                         googleSecrets: Option[ClientSecrets]) {
+
+  val localizeWithRefreshToken = googleSecrets.isDefined
+  val isDockerAuthenticated = dockerCredentials.isDefined
+}
+
 object JesAttributes {
 
-  private val keys = Set(
+  private val jesKeys = Set(
     "applicationName",
     "project",
     "baseExecutionBucket",
     "endpointUrl",
-    "authenticationMode",
     "maximumPollingInterval",
     "dockerAccount",
-    "dockerToken"
+    "dockerToken",
+    "localizeWithRefreshToken"
+  )
+
+  private val googleKeys = Set(
+    "authScheme",
+    "userAuth",
+    "accessTokenAuth",
+    "refreshTokenAuth",
+    "serviceAuth"
   )
 
   private val context = "Jes"
 
   def apply(): JesAttributes = {
     val jesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+    val googleConf = ConfigFactory.load.getConfig("google")
+    val clientSecrets = jesConf.getConfigOption("localizeWithRefreshToken") validateAny { mbConfig =>
+      mbConfig map { config => SimpleClientSecrets(config.getString("client_id"), config.getString("client_secret")) }
```

Comment:
I think you can just 

```scala
validateAny { _ map { config => ... } }
```

to avoid the wrath of those who dislike "Hungarian" naming prefixes